homeassistant/components/rpi_gpio/switch.py

```diff
@@ -59,21 +57,19 @@ def name(self):
     @property
     def should_poll(self):
         """No polling needed."""
-        return False
+        return True
 
     @property
     def is_on(self):
         """Return true if device is on."""
-        return self._state
+        return rpi_gpio.read_input(self._port) == self._on_state
```

Comment:
We don't allow blocking calls in entity properties.

Reply:
Thanks for review. I've submit a patch.